### PR TITLE
Create user input by Admin if not in system

### LINKED ISSUE #146: As an admin user I should be able to edit the netid when I create a new locker application.
### Expected behavior
As an admin user I should be able to edit the locker application with a different netid.

### Actual behavior
As an admin user when I edit the locker application with a different netid it does not persist. Every time I click another field or outside of the netid field it will change the value back to my netid. 

### Steps to replicate

https://user-images.githubusercontent.com/9905193/199262634-884d682c-0e7c-4375-aa34-4fed4eded8f5.mov

### Impact of this bug
Every application I submit as an admin has my netid. 

```
```

### PR DESCRIPTION
**Pull request by maxkadel**
Closes #146 

- user_uid is required
- tests - behavior of form is different with missing user_uid vs. missing building or other param (user_uid raises error, missing building gives message)
- Split out CAS verification into #161

### CODE REVIEW

**Comment by sandbergja** [comment]
This is a companion PR to #147.  That PR restored the old behavior when admins tried to create applications on behalf of students who didn't yet have accounts in the system (show an alert to the admin).  This PR improves on this behavior by actually creating the user account for the student if it doesn't yet exist.

File: app/controllers/locker_applications_controller.rb [hunk 1]
```diff
@@ -98,8 +100,8 @@ def lookup_user_from_params(locker_params)
     uid = locker_params.delete(:user_uid)
     user = if current_user.uid == uid
              current_user
-           else
-             User.find_by(uid: uid)
+           elsif current_user.admin?
```
Comment (sandbergja) [inline app/controllers/locker_applications_controller.rb:103]: Could you please add a test for if a non-admin attempts to pass in somebody else's uid?

File: spec/requests/locker_applications_spec.rb [hunk 2]
```diff
@@ -32,26 +32,28 @@
...
     {
+      building_id: building_one.id,
       preferred_size: 2,
       preferred_general_area: 'Preferred General Area',
       accessible: false,
       semester: 'Semester',
       status_at_application: 'Status At Application',
       department_at_application: 'Department At Application',
-      user_id: nil
+      user_uid: nil
```
Comment (sandbergja) [inline spec/requests/locker_applications_spec.rb:44]: 🎉

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks, @maxkadel !

File: app/controllers/locker_applications_controller.rb (hunk 1 above)
Comment (maxkadel) [reply app/controllers/locker_applications_controller.rb:103]: Done!

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel !

**Pull request merged**